### DISCUSSION TITLE
parameter dictionary?

### DISCUSSION THREAD

**Discussion by carolinafernandezp** [jonescompneurolab/hnn-core#512] [Q&A]
Hello,

I'm interested in understanding what the different parameters in the .param files that get created mean (especially those that cannot be directly changed through the GUI). I believe there are a total of 255 parameters. For example:

- L5Pyr_soma_gkbar_hh2: 0.06
- L5Pyr_soma_gnabar_hh2: 0.32
- L5Pyr_soma_el_hh2: -65.0

Is there a dictionary of some sort that will tell me what these different parameters mean and their units?

Thank you

**Comment by rythorpe** [comment]
Sadly, no. Most of the parameters in .param files are poorly documented and ambiguous. For the very reason you bring up, we've been trying to discourage use of .param files in hnn-core. To answer your question about those specific parameters though,

- L5Pyr_soma_gkbar_hh2 is the maximal conductance of the Hodgkin-Huxley potassium current in L5 pyramidal cell soma compartments (microsiemens, uS)
- L5Pyr_soma_gnabar_hh2 is the same, except for sodium (microsiemens, uS)
- L5Pyr_soma_el_hh2 is the reversal potential for the leak current (I believe these are in millivolts, mV)

**Comment by jasmainak** [comment]
Take a look at this:

```py
from hnn_core import jones_2009_model

net = jones_2009_model()

print(net.cell_types)
print(net.cell_types['L2_pyramidal'].sections)  # geometry of the cell compartments
print(net.cell_types['L2_pyramidal'].sections['apical_trunk'].mechs)  # mechanisms in a compartment and their parameters
```

Then to get what the parameters are actually doing, you have to look at the [mod](https://github.com/jonescompneurolab/hnn-core/tree/master/hnn_core/mod) files. For e.g., if you look at `hh2.mod` you can see the equations and what the variables mean. @carolinafernandezp if you are familiar with git, I think starting a document that describes this would be great.
